compiler/rustc_lint/messages.ftl

```diff
@@ -443,7 +443,7 @@ lint_non_local_definitions_impl = non-local `impl` definition, they should be av
             [one] `{$body_name}`
            *[other] `{$body_name}` and up {$depth} bodies
         }
-    .non_local = an `impl` definition is non-local if it is nested inside an item and neither the type nor the trait are at the same nesting level as the `impl` block
+    .non_local = an `impl` definition is non-local if it is nested inside an item and neither the type nor the trait are at the same nesting level as the `impl` block and should not impact code outside of that item
```

Comment:
that sentence seems off to me :thinking: 
```suggestion
    .non_local = an `impl` definition is non-local if it is nested inside an item and may impact type checking outside of that item. This can be the case if neither the trait or the self type are at the same nesting level as the `impl`
```
or even ignoring the second sentence, simply using " an `impl` definition is non-local if it is nested inside an item and may impact type checking outside of that item."

Reply:
The sentence also seemed a bit weird to me. Thanks for the suggestions, I ended up using the first one, since it gives a bit more details.